appyx-components/demos/common/src/commonMain/kotlin/com/bumble/appyx/components/demos/puzzle15/Puzzle15Model.kt

```diff
@@ -0,0 +1,58 @@
+package com.bumble.appyx.components.demos.puzzle15
+
+import com.bumble.appyx.components.demos.puzzle15.Puzzle15Model.Tile
+import com.bumble.appyx.interactions.Parcelable
+import com.bumble.appyx.interactions.Parcelize
+import com.bumble.appyx.interactions.RawValue
+import com.bumble.appyx.interactions.core.Element
+import com.bumble.appyx.interactions.core.asElement
+import com.bumble.appyx.interactions.core.model.transition.BaseTransitionModel
+import com.bumble.appyx.interactions.core.state.SavedStateMap
+
+class Puzzle15Model(
+    savedStateMap: SavedStateMap?
+) : BaseTransitionModel<Tile, Puzzle15Model.State>(
+    savedStateMap = savedStateMap
+) {
+    data class Tile(val value: String)
```

Comment:
I'd rather not store it as a `String`. Displaying text in the composition is only one visualisation, it could as well be a slice of a photo. Having it as an `enum` instead could benefit from exhaustive when branching in client code to choose whatever visualisation needed.